packages/amplify_datastore/ios/Classes/types/hub/FlutterHubElement.swift

```diff
@@ -46,6 +46,39 @@ public struct FlutterHubElement {
         self.lastChangedAt = hubElement.lastChangedAt
         self.deleted = hubElement.deleted
     }
+
+    init(
+        hubElement: MutationEvent,
+        modelSchemaRegistry: FlutterSchemaRegistry,
+        customTypeSchemaRegistry: FlutterSchemaRegistry,
+        modelName: String
+    ) throws {
+        do {
+            let decodedModel = try ModelRegistry.decode(modelName: modelName, from: hubElement.json) as! FlutterSerializedModel
```

Comment:
Please use `guard` - force casts will crash the app if it ever fails

Reply:
Then it fails, the force cast exception will be captured and handled at this [spot](https://github.com/HuiSF/amplify-flutter/blob/chore/ds-integration-tests-cloud/packages/amplify_datastore/ios/Classes/DataStoreHubEventStreamHandler.swift#L190-L192).

Reply:
No, that's not true. It will crash the program - unwraps are not propagated as thrown errors like they are in Dart and other languages.

<img width="1072" alt="Screen Shot 2022-03-24 at 1 13 25 PM" src="https://user-images.githubusercontent.com/24740863/160002290-2fd19a80-6a3d-449f-be45-3fa1e57d5a67.png">

Reply:
Good to know. But for this specific spot -

* `ModelRegistry.decode()` returns a `Model` otherwise it throws (would be captured)
* `FlutterSerializedModel` extends `Model`
* Within the DataStore iOS implementation, everything presented by `Model` is actually `FlutterSerializedModel`

Therefore this force unwrap should be safe here, Xcode hasn't prompt any warning as well.

Reply:
Like I said in my other comment, these are things you know but are not apparent by just reading the code. So please either write this in a comment above this line or just change it to use a `guard`/`throw` which, as you say, will have the same effect. The difference is that by changing it, it's ✨ **guaranteed** ✨ to never crash no matter what changes in our code or iOS.

You don't get a warning because you have statements that throw in the body. But your original statement was incorrect, this will crash the program if it ever fails. I'm trying to make a general point about force unwrapping, no longer specific to this one line.

Reply:
In this case, you can also create a method with this comment and reuse that throughout our DataStore code, because it sounds like we may be doing this a lot throughout and there may be something we know that the compiler doesn't.

Reply:
Yea, the native impl. could use a good clean up. I'll fine some other time to do so... or waiting for the "final big" change. I'll take your suggestion here, thanks.